### PR TITLE
Expose CNI MTU on the baremetal platform

### PR DESCRIPTION
**Pull request by knrt10**
Add docs for baremetal

Closes: #976

### CODE REVIEW

File: docs/configuration-reference/platforms/baremetal.md [hunk 1]
```diff
@@ -143,13 +145,14 @@ os_version = var.custom_default_os_version
...
 | `ssh_pubkeys`               | List of SSH public keys for user `core`. Each element must be specified in a valid OpenSSH public key format, as defined in RFC 4253 Section 6.6, e.g. "ssh-rsa AAAAB3N...". |        -         | list(string) |   true   |
 | `os_version`                | Flatcar Container Linux version to install. Version such as "2303.3.1" or "current".                                                                                         |    "current"     |    string    |  false   |
 | `os_channel`                | Flatcar Container Linux channel to install from ("flatcar-stable", "flatcar-beta", "flatcar-alpha", "flatcar-edge").                                                         | "flatcar-stable" |    string    |  false   |
-| `enable_tls_bootstrap`      | Enable TLS bootstraping for Kubelet.                                                                                                                                         |      true        |     bool     |  false   |
```
Comment (invidian) [inline docs/configuration-reference/platforms/baremetal.md:152]: Why this change?

File: pkg/platform/baremetal/template.go [hunk 2]
```diff
@@ -44,6 +44,10 @@ module "bare-metal-{{.ClusterName}}" {
   worker_macs        = {{.WorkerMacs}}
   worker_domains     = {{.WorkerDomains}}
 
+  {{- if .Config.NetworkMTU }}
```
Comment (invidian) [inline pkg/platform/baremetal/template.go:47]: `.Config` should be ommited I think.

**Review by invidian — CHANGES_REQUESTED** [review]

File: docs/configuration-reference/platforms/baremetal.md (hunk 1 above)
Comment (knrt10) [reply docs/configuration-reference/platforms/baremetal.md:152]: This is automatically done on formatting.

**Review by invidian — DISMISSED** [review]
ok

**Review by invidian — APPROVED** [review]

**Review by surajssd — APPROVED** [review]
LGTM

**Review by iaguis — APPROVED** [review]
